modules/serverpod_auth/serverpod_auth_server/lib/src/business/email_auth.dart

```diff
@@ -81,11 +115,22 @@ class Emails {
   /// migrated. If null, all entries in the database will be migrated.
   ///
   /// Returns the number of migrated entries.
+  ///
+  /// Warning: This migration method is designed for password hashes generated
+  /// by the framework's default algorithm. Hashes stored with a custom
+  /// generator or different algorithm may produce unexpected results.
   static Future<int> migrateLegacyPasswordHashes(
     Session session, {
     int batchSize = 100,
     int? maxMigratedEntries,
   }) async {
+    if (AuthConfig.current.passwordHashGenerator.hashCode !=
+            defaultGeneratePasswordHash.hashCode ||
+        AuthConfig.current.passwordHashValidator.hashCode !=
+            defaultValidatePasswordHash.hashCode) {
+      throw Exception(
+          'Can\'t migrate legacy password with another password hash algorithm');
+    }
```

Comment:
Suggestion: Change to `Legacy password hash migration not supported when using custom password hash algorithm.`